analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/FinallyPoint.cs

```diff
@@ -0,0 +1,45 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using SonarAnalyzer.CFG.Roslyn;
+
+namespace SonarAnalyzer.SymbolicExecution.Roslyn
+{
+    internal sealed class FinallyPoint
+    {
+        private readonly ControlFlowBranch branch;
+        private readonly int finallyIndex;
+
+        public bool IsFinallyBlock => finallyIndex < branch.FinallyRegions.Length;
+        public int BlockIndex => IsFinallyBlock ? branch.FinallyRegions[finallyIndex].FirstBlockOrdinal : branch.Destination.Ordinal;
```

Comment:
So we now consider the `finally` regions as blocks that appear before any other destination block.

Reply:
As a big picture idea, yes. 

(while keeping in mind that the finally region can contain multiple blocks)